### PR TITLE
Fix applications route to render application by id

### LINKED ISSUE #783: show user application by it's id
### Issue Description
If super pass id=<applicationid> in query params in application route then it render the particular application to UI 

### Reproducibility

 - [ ] This issue is reproducible
 - [x] This issue is not reproducible

### Severity/Priority
 - [ ] Critical
 - [x] High
 - [ ] Medium
 - [ ] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by vinit717**
## Issue Ticket Number
- closes #783 

## Description

In this PR super user can view the particular application by passing in the id of application in query params

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [ ] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

image
image

https://github.com/user-attachments/assets/46cafae6-a174-46d3-805d-9ea9c164fd70

## Test Coverage
<details>
<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

File: applications/script.js [hunk 1]
```diff
@@ -252,27 +253,63 @@ async function renderApplicationCards(next, status, isInitialRender) {
...
+  noApplicationFoundText.classList.add('hidden');
+  changeLoaderVisibility({ hide: false });
+  isDataLoading = true;
+
+  try {
+    const data = await getApplicationById(id);
+
+    const application = data.application;
+
+    isDataLoading = false;
```
Comment (iamitprakash) [inline applications/script.js:294]: don't we need to handle this flag in catch block

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: applications/utils.js [hunk 2]
```diff
@@ -40,6 +37,30 @@ async function getApplications({ applicationStatus, size = 5, next = '' }) {
   }
 }
 
+async function getApplicationById(applicationId) {
```
Comment (iamitprakash) [inline applications/utils.js:40]:
```
async function getApplicationById(applicationId) {
  try {
    const res = await fetch(`${BASE_URL}/applications/${applicationId}`, {
      method: 'GET',
      credentials: 'include',
      headers: {
        'Content-type': 'application/json',
      },
    });

    const data = await res.json();

    if (!res.ok) {
      throw new Error(data.message || 'Failed to fetch application');
    }

    console.log('data', data);
    return data;
  } catch (error) {
    console.error('Error in getApplicationById:', error);
    throw error;
  }
}
```

File: applications/utils.js [hunk 3]
```diff
@@ -1,7 +1,4 @@
-const BASE_URL =
-  window.location.hostname === 'localhost'
-    ? 'https://staging-api.realdevsquad.com'
-    : window.API_BASE_URL;
+const BASE_URL = 'http://localhost:3000';
```
Comment (yesyash) [inline applications/utils.js:1]: - should this change be reverted?

**Review by yesyash — CHANGES_REQUESTED** [review]

File: applications/utils.js [hunk 4]
```diff
@@ -40,6 +37,30 @@ async function getApplications({ applicationStatus, size = 5, next = '' }) {
...
+    });
+
+    if (!res.ok) {
+      const error = await res.json();
+      throw error;
+    }
+
+    const data = await res.json();
+
+    console.log('data', data);
```
Comment (yesyash) [inline applications/utils.js:57]: remove this

File: applications/script.js [hunk 5]
```diff
@@ -252,27 +253,63 @@ async function renderApplicationCards(next, status, isInitialRender) {
...
+    const applications = data.applications;
+    nextLink = data.next;
+
+    if (isInitialRender) filterButton.classList.remove('hidden');
+    if (!applications.length)
+      return noApplicationFoundText.classList.remove('hidden');
+
+    applicationContainer.innerHTML = '';
+    applications.forEach((application) => {
+      console.log('applications', application);
```
Comment (yesyash) [inline applications/script.js:271]: remove this

**Review by yesyash — CHANGES_REQUESTED** [review]

File: applications/script.js (hunk 1 above)
Comment (vinit717) [reply applications/script.js]: fixed

File: applications/utils.js (hunk 3 above)
Comment (vinit717) [reply applications/utils.js]: fixed

File: applications/script.js (hunk 5 above)
Comment (vinit717) [reply applications/script.js:271]: fixed

File: applications/utils.js (hunk 4 above)
Comment (vinit717) [reply applications/utils.js]: fixed

File: applications/script.js [hunk 6]
```diff
@@ -271,8 +272,33 @@ async function renderApplicationCards(next, status, isInitialRender) {
...
+    applicationContainer.classList.add('center');
+  } catch (error) {
+    console.error('Error fetching application by user ID:', error);
+    noApplicationFoundText.classList.remove('hidden');
+  } finally {
+    isDataLoading = false;
+    changeLoaderVisibility({ hide: true });
+  }
+}
+
```
Comment (Achintya-Chatterjee) [inline applications/script.js:298]: error handling can be improved

**Pull request merged**